### ISSUE TITLE
Add interface for EV3 buttons

### ISSUE THREAD

**Issue by WasabiFan** [WasabiFan/ev3dev-lang-js#13]

**Comment by massimiliano-mantione** [comment]
Is this hard?

I am working on a project, I started using the nodejs bindings because the project fits the "async" paradigm well, I know Javascript very well and I never used Python, but just now I noticed that button support is missing.

What would be needed to add button support?

**Comment by WasabiFan** [comment]
No, this should not be hard; I have written some proof-of-concept Node.js code previously that read the button states. The main reason that I haven't done it yet is that it requires polling to detect changes in button states, and I didn't previously have infrastructure in the library to do that cleanly. I do have such infrastructure now, so adding buttons should be reasonably easy; I can go find the work I have done so far in a few hours. I'll post some example code here so that you are able to use the buttons without waiting for me if your project is time-sensitive,

**Comment by WasabiFan** [comment]
I've uploaded my button demo [here](https://gist.github.com/WasabiFan/51a395564cf5938cd2dc9d5885bad68d). The code watches the input event file and asynchronously parses and handles events as they are received. I plan to integrate this functionality into the library soon, but until then feel free to adapt my example to your needs. I added some comments to try to explain what's going on; feel free to ask if the code is unclear or something isn't working correctly.

The code makes some assumptions about the platform that it's running on (e.g. that it's little endian) but should work on all the ev3dev-supported platforms.
